File: Server/src/main/java/com/api/server/foodb/OntologyTerm.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "ontology_terms")
public class OntologyTerm {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Size(max = 255)
    @Column(name = "term")
    private String term;

    @Lob
    @Column(name = "definition")
    private String definition;

    @Size(max = 255)
    @Column(name = "external_id")
    private String externalId;

    @Size(max = 255)
    @Column(name = "external_source")
    private String externalSource;

    @Size(max = 255)
    @Column(name = "comment")
    private String comment;

    @Size(max = 255)
    @Column(name = "curator")
    private String curator;

    @Column(name = "parent_id")
    private Integer parentId;

    @Column(name = "level")
    private Integer level;

    @NotNull
    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @NotNull
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Column(name = "legacy_id")
    private Integer legacyId;

}

File: DotNetServer/DotNetServer/Entities/Foodb/OntologyTerm.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DotNetServer.Entities.Foodb;

[Table("ontology_terms")]
public partial class OntologyTerm
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("term")] [StringLength(255)] public string? Term { get; set; }

    [Column("definition", TypeName = "text")]
    public string? Definition { get; set; }

    [Column("external_id")]
    [StringLength(255)]
    public string? ExternalId { get; set; }

    [Column("external_source")]
    [StringLength(255)]
    public string? ExternalSource { get; set; }

    [Column("comment")]
    [StringLength(255)]
    public string? Comment { get; set; }

    [Column("curator")]
    [StringLength(255)]
    public string? Curator { get; set; }

    [Column("parent_id")] public int? ParentId { get; set; }

    [Column("level")] public int? Level { get; set; }

    [Column("created_at", TypeName = "datetime")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime UpdatedAt { get; set; }

    [Column("legacy_id")] public int? LegacyId { get; set; }

    [InverseProperty("OntologyTerm")]
    public virtual ICollection<OntologySynonym> OntologySynonyms { get; set; } = new List<OntologySynonym>();
}